src/relay/backend/vm/compiler.cc

```diff
@@ -1041,13 +1013,31 @@ IRModule VMCompiler::OptimizeModule(IRModule mod, const TargetsMap& targets_arg,
 
   Array<Pass> pass_seqs = relay::backend::GetPassPrefix(targets, true);
 
+  Device default_device;
```

Comment:
Is it possible for us to compute this outside the OptimizeModule, ideally this should just build the optimization pipeline and have minimal logic.

Reply:
Alas it's still there, I've run out of steam and we should centralize this and the (now reigged) default device logic in build_module.cc